src/internal/packager2/layout/package.go

```diff
@@ -179,12 +179,30 @@ func (p *PackageLayout) GetImageDir() string {
 }
 
 func (p *PackageLayout) Archive(ctx context.Context, dirPath string, maxPackageSize int) error {
-	packageName := fmt.Sprintf("%s%s", sources.NameFromMetadata(&p.Pkg, false), sources.PkgSuffix(p.Pkg.Metadata.Uncompressed))
+	// Ensure output path exists
+	// NOTE(mkcp): We may want to move this behavior to archive.Compress
```

Comment:
Yeah I tested it out and on previous version of Zarf users could run commands like `zarf tools archiver compress dir-to-compress dir-not-exists/my-file.tar.zst` so I think the behavior should live there

Reply:
Makes sense!